docs/DeveloperGuide.md

```diff
@@ -1,38 +1,1061 @@
+  - [Target user profile](#target-user-profile)
+  - [Value proposition](#value-proposition)
+- [User Stories](#user-stories)
+- [Non-Functional Requirements](#non-functional-requirements)
+- [Acknowledgements](#acknowledgements)
+  - [Documentation](#documentation)
+  - [Storage](#storage-1)
+  - [Unit Tests](#unit-tests-1)
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+# Design
 
-## Design & implementation
+## Architecture
+![PocketPal Architecture](./static/PocketPalArchitecture.png)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The diagram above illustrates the high-level overview of `PocketPal`.
 
+__Main Components of PocketPal__
+- The entry point to our application is `pocketpal.PocketPal`, which creates the `Frontnend` and `Backend` instances at launch.
+- `PocketPal` also sets up logging for the application, which outputs the logs to `logs/logging.txt`.
 
-## Product scope
-### Target user profile
+__Frontend__
+`Frontend` is made up of the following components
+- `UI`: Handles display of user interface
+- `Parser`: Converts text entered by users into `Command` if valid
+- `Command`: Contains the instructions to be executed as input by the user
+
+__Backend__
+`Backend` is made up of the following components
+- `Endpoint`: The parent component used by various endpoints (`EntryEndpoint` and `EntriesEndpoint`), where the frontend can send a `Request` 
+- `EntryLog`: The main data structure used for temporarily storing multiple `Entry` objects
+- `Storage`: Reads data from, and writes data to the disk, which can be found at `data/storage.txt`
+
+__Communication between `Frontend` and `Backend`__
+- PocketPal uses the `Gson` library to serialise and deserialise objects where applicable, such as `Entry` and `EntryLog`.
+- This allows us to standardise the communication framework between `Frontend` and `Backend`, by using a simplified HTTP model.
+- Each endpoint takes in a `Request`, and returns a `Response` based on the requested data.
+
+## Frontend
+
+<!-- @@author adenteo -->
+### Parser
+
+The `Parser` class is a fundamental component instantiated as soon as PocketPal is initialised. Its __purpose__ is to
+convert the user's input into structured data which then produces clear instructions for the rest of the program.
+
+Some of its core features include:
+
+- Breaking down user input and extracting the relevant data for further processing.
+- Performing input validation and error handling to ensure that input data is in the correct format and ready to be
+  processed.
+- Converting the input data into the correct format and returning it as a `Command` class to be further processed by the
+  application.
+
+Here's a class diagram that shows the core structure of the `Parser` class.
+
+![ParserClassDiagram](static/frontend/parser/ParserClassDiagram.png)
```

Comment:
The class diagram does not help in clarifying the parser mechanism as it shows only the methods in Parser. The explanation below and sequence diagram are sufficient.